primefaces/src/main/java/org/primefaces/component/api/InputAware.java

```diff
@@ -0,0 +1,240 @@
+    @FacesBehaviorEvent(name = "mouseout", event = AjaxBehaviorEvent.class, description = "Fires when the mouse leaves the element."),
+    @FacesBehaviorEvent(name = "mouseover", event = AjaxBehaviorEvent.class, description = "Fires when the mouse enters the element."),
+    @FacesBehaviorEvent(name = "mouseup", event = AjaxBehaviorEvent.class, description = "Fires when a mouse button is released over the element."),
+    @FacesBehaviorEvent(name = "paste", event = AjaxBehaviorEvent.class, description = "Fires when a paste operation occurs on the element."),
+    @FacesBehaviorEvent(name = "reset", event = AjaxBehaviorEvent.class, description = "Fires when the form is reset."),
+    @FacesBehaviorEvent(name = "scroll", event = AjaxBehaviorEvent.class, description = "Fires when the element is scrolled."),
+    @FacesBehaviorEvent(name = "search", event = AjaxBehaviorEvent.class, description = "Fires when a search is initiated on the element."),
+    @FacesBehaviorEvent(name = "select", event = AjaxBehaviorEvent.class, description = "Fires when some text is selected in the element."),
+    @FacesBehaviorEvent(name = "valueChange", event = AjaxBehaviorEvent.class, description = "Fires when the element's value is changed.", defaultEvent = true),
+    @FacesBehaviorEvent(name = "wheel", event = AjaxBehaviorEvent.class, description = "Fires when the wheel event is triggered on the element.")
+})
+public interface InputAware {
+
+    @Property(description = "Specifies a shortcut key to activate or focus the element.")
+    String getAccesskey();
+
+    @Property(description = "Identifies the element(s) that describe the object, enhancing accessibility.")
+    String getAriaDescribedBy();
+
+    @Property(description = "Specifies a converter instance to be used with the component.")
+    @SuppressWarnings("rawtypes")
+    Converter getConverter();
+
+    @Property(description = "Custom message to display when conversion fails.")
+    String getConverterMessage();
+
+    @Property(description = "Indicates whether the input is disabled.", defaultValue = "false")
+    boolean isDisabled();
+
+    @Property(description = "Whether the component should process its value immediately in the Apply Request Values phase.", defaultValue = "false")
+    boolean isImmediate();
+
+    @Property(description = "Specifies the type of data expected in the input, aiding virtual keyboards.")
+    String getInputmode();
+
+    @Property(description = "Defines a label for the input element.")
+    String getLabel();
+
+    @Property(description = "Specifies the language of the element's content.")
+    String getLang();
+
+    @Property(description = "Indicates whether the local value of the component has been set.", defaultValue = "false")
+    boolean isLocalValueSet();
+
+    @Property(description = "Defines the maximum number of characters allowed in the input.", defaultValue = "-2147483648")
```

Comment:
Integer.MIN_VALUE has to be expressed as -2147483648 since defaultValue requires a String?  didn't know if I coudl do Integer.MIN_VALUE.toString()?

Reply:
Nope `Integer.MIN_VALUE.toString()` not allowed but no big deal.